### PR TITLE
Test for cyclic dependency added to should_fail

### PR DESCRIPTION
**Pull request by kayagokalp**
This adds a test to the `should_fail` part of the suite so that once there is a cyclic dependency we are indeed not building (more precisely not trying to build, i.e infinite fetching) the project.

I added a `dependency_a` which depends on `dependency_b`. But `dependency_b` depends on `dependency_a`. So trying to build `dependency_a` (or `dependency_b`) should fail.

### CODE REVIEW

File: test/src/e2e_vm_tests/test_programs/should_fail/cyclic_dependency/dependency_a/Forc.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[project]
+authors = ["Fuel Labs <contact@fuel.sh>"]
+entry = "lib.sw"
+license = "Apache-2.0"
+name = "dependency_a"
```
Comment (mitchmindtree) [inline test/src/e2e_vm_tests/test_programs/should_fail/cyclic_dependency/dependency_a/Forc.toml:5]:
```suggestion
name = "dependency_a"
implicit-std = false
```

File: test/src/e2e_vm_tests/test_programs/should_fail/cyclic_dependency/dependency_b/Forc.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[project]
+authors = ["Fuel Labs <contact@fuel.sh>"]
+entry = "lib.sw"
+license = "Apache-2.0"
+name = "dependency_b"
```
Comment (mitchmindtree) [inline test/src/e2e_vm_tests/test_programs/should_fail/cyclic_dependency/dependency_b/Forc.toml:5]:
```suggestion
name = "dependency_b"
implicit-std = false
```

**Review by mitchmindtree — CHANGES_REQUESTED** [review]
Looks good, just missing a couple of `implicit-std = false` (you can spot this by seeing the `git` dependency show up in the lock files).

**Comment by mitchmindtree** [comment]
@kayagokalp would you mind also rebuilding those two packages locally in order to update their lock files and push the updated lock files too? Otherwise they'll end up needing to be updated in some future unrelated PR. #1951 aims at addressing this.

**Comment by kayagokalp** [comment]
Oh right, my bad. Pushing those as well

**Comment by mitchmindtree** [comment]
I think it's fine to include it still, and we'll need to as of #1951 anyway.

The error is showing up in the E2E test CI pass, but it looks like it's `panic!`ing rather that just exiting with the error. I'll do up a quick PR for this.

**Comment by kayagokalp** [comment]
Hmm, I might be missing something but I think cannot add the lock files since the lock file of `dependency_a` or `dependency_b` is not actually being generated at the moment (because we cannot build them). 

The old lock file that was present in this PR was generated before I added `dependency_b` as a dependency for `dependency_a` (i.e, before creating the cyclic dependency) and I accidentally pushed that file.

**Comment by mitchmindtree** [comment]
> Hmm, I might be missing something but I think cannot add the lock files since the lock file of dependency_a or dependency_b is not actually being generated at the moment (because we cannot build them).

:laughing: that's a really good point! Turns out the aforementioned CI error was unrelated to this and solved in #1955 anyways.

**Review by mitchmindtree — APPROVED** [review]

**Pull request merged**